cypress/integration/external/user-returns-view.spec.js

```diff
@@ -0,0 +1,62 @@
+
+const { setUp, tearDown } = require('../../support/setup');
+// const checkErrorMessage = (message) => {
+/* describe('it sees the correct error message', () => {
+  cy.get('.govuk-list > li > a').should('have.text');
+  // });
+}); */
+
+describe('check for different return status as an external user', () => {
+  beforeEach(() => {
+    tearDown();
+    setUp('barebones');
+  });
+
+  afterEach(() => {
+    tearDown();
+  });
+
+  it('sees the page title', () => {
+    cy.visit(Cypress.env('USER_URI'));
+
+    // tap the sign in button on the welcome page
+    cy.get('a[href*="/signin"]').click();
+    cy.fixture('users.json').then(users => {
+      cy.get('input#email').type(users.external);
+    });
+    cy.get('input#password').type(Cypress.env('DEFAULT_PASSWORD'));
+
+    //  Click Sign in Button
+    cy.get('.govuk-button.govuk-button--start').click();
+    cy.contains('Your licences').should('have.class', 'govuk-heading-l');
+
+    //  Search for the licence
+    describe('sees the licences table', () => {
+      cy.contains('AT/CURR/MONTHLY/02').should('be.visible');
+      cy.get('#results').should('be.visible');
+    });
+
+    describe('sees the three licences created by the setup routine', () => {
+      cy.get('#results').should('contain.text', 'AT/CURR/MONTHLY/02');
+    });
+    describe('clicks on the MONTHLY licence 02', () => {
+      cy.contains('AT/CURR/MONTHLY/02').click();
+      cy.get('.govuk-heading-l').should('be.visible').and('contain.text', 'Licence number AT/CURR/MONTHLY/02');
+    });
+
+    // Click on returns to see different status
+    describe('sees the Summary table', () => {
+      cy.get('#summary').should('be.visible');
+    });
+    describe('it clicks on the returns tab link', () => {
+      cy.get('#tab_returns').click();
+    });
+    describe('sees the returns table', () => {
+      cy.get('#returns').should('be.visible');
+    });
+    describe('sees the status column', () => {
+      cy.get(':nth-child(1) > :nth-child(4) > .govuk-tag').should('be.visible').and('contain.text', 'Due');
+      cy.get(':nth-child(2) > :nth-child(4) > .govuk-tag').should('be.visible').and('contain.text', 'Overdue');
+    });
```

Comment:
I think it is better to be more explicit with what you look for in the page using the tbody tag just incase the header or somewhere else in the page another govuk-tag is added. 
Also we should check only the correct data is visible to external user by checking there are only 4 rows and checking the status for each row becuase you do not check for the completed return tag.
For this licence we know there are some void returns so perhaps doing an explicit positive check will be better or you could check if the void status tag is not found? 
i.e. 
```
 describe('see the returns table with 4 rows', () => {
      cy.get('tbody').find('tr').should('have.length', 4)
      describe('each row has the correct status tag', () => {
      cy.get('tbody').find(':nth-child(1) > :nth-child(4) > .govuk-tag').should('be.visible').and('contain.text', 'Due');
      cy.get('tbody').find(':nth-child(2) > ... 'Overdue');
      cy.get('tbody').find(':nth-child(3) > ... 'Complete');
      cy.get('tbody').find(':nth-child(4) > ... 'Overdue');
      });
    });
```